Make error pages translatable

tomhughes opened the pull request:
Fixes #3877

gravitystorm commented on config/locales/en.yml:
```diff
@@ -572,6 +572,19 @@ en:
           notice: Application Registered.
+  errors:
+    contact:
+      html: Feel free to <a href="http://wiki.openstreetmap.org/wiki/Contact" title="Various contact channels explained">contact</a> the OpenStreetMap community if you have found a broken link / bug. Make a note of the exact URL of your request.
```
We should avoid including raw html tags in translations - these can be hard for translators to get right, among other things. Instead I think it's better to have each component of the paragraph as separate keys, e.g.
```
errors:
  contact:
    contact_url: http:/...
    contact_url_title: Various...
    contact: contact
    contact_the_community_html: Feel free to %{contact_link} the OpenStreetMap ....
```

... and the appropriate `link_to` when calling the translation.

tomhughes replied:
Done.

gravitystorm commented on config/locales/en.yml:
```diff
@@ -572,6 +572,19 @@ en:
           notice: Application Registered.
+  errors:
+    contact:
+      html: Feel free to <a href="http://wiki.openstreetmap.org/wiki/Contact" title="Various contact channels explained">contact</a> the OpenStreetMap community if you have found a broken link / bug. Make a note of the exact URL of your request.
```
`https` for the link too

tomhughes replied:
Done.

gravitystorm commented on config/locales/en.yml:
```diff
@@ -572,6 +572,19 @@ en:
+    internal_server_error:
+      title: Application error
+      paragraph_1: The OpenStreetMap server encountered an unexpected condition that prevented it from fulfilling the request (HTTP 500)
+      paragraph_2: This may be a problem in our Ruby On Rails code. 500 occurs with exceptions thrown outside of an action (like in Dispatcher setups or broken Ruby code)
```
I think the second sentence here is jargon that we shouldn't show to end users. I mention it here because I don't know what "Dispatcher setups" are, in this context, and so I wouldn't be able to advise a translator what it means either! :smile:

tomhughes replied:
Agreed... It's not really helpful so I've removed it.

gravitystorm commented on config/locales/en.yml:
```diff
@@ -572,6 +572,19 @@ en:
+      html: Feel free to <a href="http://wiki.openstreetmap.org/wiki/Contact" title="Various contact channels explained">contact</a> the OpenStreetMap community if you have found a broken link / bug. Make a note of the exact URL of your request.
+    forbidden:
+      title: Forbidden
+      paragraph_1: The operation you requested on the OpenStreetMap server is only available to administrators (HTTP 403)
```
Eh, not strictly true, but whatever.

tomhughes commented:
Apologies for merging this - it wasn't my intention and I must have pushed to the wrong place. I assume you're happy with it now I addressed your comments?

gravitystorm commented:
No worries - I was happy with the updated version!
